src/componentsNext/NotificationMenu/index.tsx

```diff
@@ -0,0 +1,82 @@
+import {
+  Notification,
+  NotificationClose,
+  NotificationLoading,
+  NotificationMenuSeparator,
+  NotificationMessage,
+  NotificationTitle,
+} from "~/design/Notification/";
+
+import { Check } from "lucide-react";
+import { Link } from "react-router-dom";
+import { twMergeClsx } from "~/util/helpers";
+import { useNamespaceLinting } from "~/api/namespaceLinting/query/useNamespaceLinting";
+import { useNotificationConfig } from "./config";
+import { useTranslation } from "react-i18next";
+
+interface NotificationMenuProps {
+  className?: string;
+}
+
+const NotificationMenu: React.FC<NotificationMenuProps> = ({ className }) => {
+  const { t } = useTranslation();
+  const { data, isLoading } = useNamespaceLinting();
+  const notificationConfig = useNotificationConfig();
+  const showIndicator = !!data?.issues.length;
+
+  const possibleNotifications = Object.entries(notificationConfig ?? {});
+
+  return (
+    <div className={twMergeClsx("self-end text-right", className)}>
+      <Notification showIndicator={showIndicator}>
+        <NotificationTitle>
+          {t("components.notificationMenu.title")}
+        </NotificationTitle>
+        <NotificationMenuSeparator />
+        {isLoading && (
+          <NotificationLoading>
+            {t("components.notificationMenu.isLoading.description")}
+          </NotificationLoading>
+        )}
+        {showIndicator ? (
+          possibleNotifications.map(
+            ([notificationType, notificationConfig], index, srcArr) => {
+              const isLastListItem = index === srcArr.length - 1;
+              const matchingNotification = data.issues.filter(
```

Comment:
```suggestion
              const matchingNotifications = data.issues.filter(
```
Arrays should be named in plural.